bundles/org.openhab.ui/web/src/assets/definitions/blockly/blocks-color.js

```diff
@@ -0,0 +1,54 @@
+/*
+* Adds new blocks to the colour sections
+*/
+
+import Blockly from 'blockly'
+
+export default function (f7) {
+  /*
+  * converts a hex color string in to an openHAB hue-saturation-brightness string
+  * Block
+  */
+  Blockly.Blocks['oh_color_to_hsb'] = {
+    init: function () {
+      this.appendValueInput('hexColor')
+        .appendField('hsb of')
+        .setCheck('Colour')
+      this.setInputsInline(false)
+      this.setOutput(true, 'String')
+      this.setColour(0)
```

Comment:
To keep the block the same color as the rest in this category (preferred):
```suggestion
      this.setColour('%{BKY_COLOUR_HUE}')
```
cf https://developers.google.com/blockly/guides/create-custom-blocks/block-colour#colour_references

Reply:
Learned something new again - thx for the hint.

Reply:
@ghys I committed this "this.setColour('%{BKY_COLOUR_HUE}')" here but I also want to add the shadow block (while you can work on the main branch, I have to do everything on the fork)

To not run into some git trouble again (I definitely want to learn that), what is the right approach?

Here's what I would do 

- fetch upstream  on my repo https://github.com/stefan-hoehn/openhab-webui on github
![image](https://user-images.githubusercontent.com/5937600/148105685-69833629-a5b5-422f-85e9-e697b3f59f5d.png)
- should I do a "fetch upstream origin/main" locally as well?
- then "git pull" locally
- "git rebase upstream/main" locally 
- git commit of my shadow block (but that adds another commit, doesn't it?
- git push

Reply:
1. `git pull --rebase`
2. if git says "you have uncommitted changes", do a `git stash` before and try 1. again
3. `git stash pop` if you had to do the git stash to recover your pending changes